data/crac/crac-impl/src/test/java/com/farao_community/farao/data/crac_impl/json/CracImportExportTest.java

```diff
@@ -72,46 +72,40 @@ public void cracTest() {
 
         simpleCrac.addNetworkElement(new NetworkElement("pst"));
         simpleCrac.addNetworkElement(new NetworkElement("injection"));
-        simpleCrac.addNetworkAction(new PstSetpoint("pstSetpointId", "pstSetpointName", "RTE", usageRules, simpleCrac.getNetworkElement("pst"), 15, CENTERED_ON_ZERO));
 
-        Set<AbstractElementaryNetworkAction> elementaryNetworkActions = new HashSet<>();
-        PstSetpoint pstSetpoint = new PstSetpoint(
-                "pstSetpointId",
-                "pstSetpointName",
-                "RTE",
-                new ArrayList<>(),
+        // network action with one pst set point
+        PstSetpointImpl pstSetpoint1 = new PstSetpointImpl(
+            simpleCrac.getNetworkElement("pst"),
+            15,
+            CENTERED_ON_ZERO);
+
+        simpleCrac.addNetworkAction(new NetworkActionImpl("pstSetpointRaId", "pstSetpointRaName", "RTE", usageRules, Collections.singleton(pstSetpoint1)));
+
+        // complex network action with one pst set point and one topology
+        PstSetpointImpl pstSetpoint2 = new PstSetpointImpl(
                 simpleCrac.getNetworkElement("pst"),
                 5,
-                CENTERED_ON_ZERO
-        );
-        Topology topology = new Topology(
-                "topologyId",
-                "topologyName",
-                "RTE",
-                new ArrayList<>(),
+                CENTERED_ON_ZERO);
+
+        TopologicalActionImpl topology = new TopologicalActionImpl(
                 simpleCrac.getNetworkElement("neId"),
-                ActionType.CLOSE
-        );
-        elementaryNetworkActions.add(pstSetpoint);
-        elementaryNetworkActions.add(topology);
-        ComplexNetworkAction complexNetworkAction = new ComplexNetworkAction(
+                ActionType.CLOSE);
+
+        NetworkActionImpl complexNetworkAction = new NetworkActionImpl(
                 "complexNetworkActionId",
                 "complexNetworkActionName",
                 "RTE",
-                new ArrayList<>(),
-                elementaryNetworkActions
-        );
+                Collections.singletonList(new FreeToUseImpl(UsageMethod.AVAILABLE, initialInstant)),
+                new HashSet<>(Arrays.asList(pstSetpoint2, topology)));
```

Comment:
You can use `Set.of(...)`

Reply:
didn't know that method. it's indeed better